src/parser/column_definition.cpp

```diff
@@ -49,6 +72,11 @@ void ColumnDefinition::Serialize(Serializer &serializer) const {
 		writer.WriteOptional(default_value);
 	}
 	writer.WriteField<TableColumnType>(category);
+	if (is_serial) {
+		writer.WriteString("T");
+	} else {
+		writer.WriteString("F");
+	}
```

Comment:
this is a bit ugly, but a quick way to fix this without implementing additional methods when serializing and deserializing, it slightly bothers me that we are losing space but I am prioritizing getting something usable here rather than saving a few bits

Reply:
I thought I commented this before, but I think my comment got lost when I refreshed or something.

Would it be a good idea to add serial as a `TableColumnType`? The column type is already serialized, and adding this breaks backward compatibility with storage I think.

Reply:
good idea, but I am not 100% sure semantically this would make sense? changing something as simple as the data type would then involve changing the table column type. also I assume the whole column type implementation would have to be majorly rewritten when implementing stored columns (which should be done fairly soon if we want to have an up-to-date SERIAL implementation and I would happily do if it's manageable as a task for my poor lil brain). I assume when you say backwards compatibility you are only talking about de/serialization - perhaps serial can be implemented as optional field instead?

Reply:
You could throw an error if you don't want to implement all that, virtual generated columns don't support changing the type either:
```sql
D alter table test alter column j set type integer;
Error: Not implemented Error: Changing types of generated columns is not supported yet
```

Or, if you add the serial field as an optional field at the end of the (de)serialization, it won't break backwards compatibility.